eng/ShowNoProjectToBuildError.proj

```diff
@@ -0,0 +1,16 @@
+<Project>
+
+  <!--
+    Display an error if no projects will be built. Echo back the Subset variable if set: a typo in
+    that variable is the only anticipated way to hit this situation.
+  -->
+  <Target Name="Build">
+    <PropertyGroup>
+      <Message>No ProjectToBuild items found</Message>
+      <Message Condition="'$(Subset)' != ''">$(Message) in subset matching '$(Subset)'</Message>
```

Comment:
It is a little unfortunate that a typo of a single subset provides a meaningful error UX but there isn't one when specifying a set - e.g. corehost-tast (typo of corehost-test)

Reply:
Yeah... I'll see if I can do this better. One limitation is that right now I don't have a list of the valid subsets, but I think it's worthwhile to add that.